### PR TITLE
Make k8s v1.10.8 (was: v1.11.0) available by default

### PR DESCRIPTION
**Pull request by alex-leonhardt**
Enables v1.11.0 and sets it to be the default version.

### CODE REVIEW

**Comment by alex-leonhardt** [comment]
I inline updated the 2 files, which is probably why the CLA doesn't match... 

/assign @jimmidyson

**Comment by alex-leonhardt** [comment]
I signed it

File: deploy/minikube/k8s_releases.json [hunk 1]
```diff
@@ -101,4 +104,4 @@
   {
     "version": "v1.3.0"
   }
-]
```
Comment (dlorenc) [inline deploy/minikube/k8s_releases.json]: nit: can you add the newline back?

**Comment by dlorenc** [comment]
@minikube-bot OK to test

**Comment by alex-leonhardt** [comment]
Hmm - not sure how the version bump broke this 

```
Sep 17 22:02:36 kvm-integration-slave kubelet[19302]: I0917 22:02:36.265112   19302 feature_gate.go:230] feature gates: &{map[]}
Sep 17 22:02:36 kvm-integration-slave kubelet[19302]: I0917 22:02:36.265410   19302 plugins.go:97] No cloud provider specified.
Sep 17 22:02:36 kvm-integration-slave kubelet[19302]: F0917 22:02:36.265605   19302 server.go:262] failed to run Kubelet: unable to load bootstrap kubeconfig: stat /etc/kubernetes/bootstrap-kubelet.conf: no such file or directory
```

Above is the last failures I saw from the log on `Linux-None`.

**Comment by alex-leonhardt** [comment]
Just saw 
```
		[init] using Kubernetes version: v1.11.0
		[preflight] running pre-flight checks
			[WARNING FileExisting-socat]: socat not found in system path
		I0917 22:02:34.410548   19189 kernel_validator.go:81] Validating kernel version
		I0917 22:02:34.410624   19189 kernel_validator.go:96] Validating kernel config
			[WARNING SystemVerification]: docker version is greater than the most recently validated version. Docker version: 17.07.0-ce. Max validated version: 17.03
			[WARNING Hostname]: hostname "minikube" could not be reached
			[WARNING Hostname]: hostname "minikube" lookup minikube on 169.254.169.254:53: no such host
		[preflight] Some fatal errors occurred:
			[ERROR FileExisting-crictl]: crictl not found in system path
``` 
the actual error, I guess, is .. `[ERROR FileExisting-crictl]: crictl not found in system path`, in the Linux-None test output ... (line 235, 236) and
```
		E0917 22:02:35.007986   19147 start.go:297] Error starting cluster:  kubeadm init error 
``` 
on line 219 though that's just out of order, the above errors happen before this one.

Hope that helps somehow?

**Comment by afbjorklund** [comment]
@alex-leonhardt : you don't have to update `k8s_releases.json`, since it is only used for localkube (not for kubeadm) and only *after* it has been uploaded to the GCS location and made available for download...

Most likely, that file should be deleted now. 

But the about socat/crictl errors are weird.

**Comment by afbjorklund** [comment]
Hmm, `Docker version: 17.07.0-ce` doesn't _sound_ like the minikube.iso.
Maybe running on the localhost, without recommended packages installed ?

Promoting `crictl` from a WARNING to an ERROR is new, see: https://github.com/kubernetes/kubeadm/issues/942
Now (1.11+) **everyone** needs to install "cri-tools", before running "kubeadm"


Unfortunately this is mostly* undocumented for anything but Container Linux:
https://kubernetes.io/docs/setup/independent/install-kubeadm/#before-you-begin

* added as a package dependency in https://github.com/kubernetes/release/pull/573

It is normally downloaded from https://github.com/kubernetes-sigs/cri-tools
For the minikube.iso, we build it from source (currently version `v1.11.1`)

**Comment by alex-leonhardt** [comment]
Is there anything I can do, or do we need to wait for #3150 to be completed?

**Comment by afbjorklund** [comment]
@alex-leonhardt : probably better to move to `v1.10.8` as the **default** version, while minikube support for 1.11 and 1.12 stabilizes ? Users can always chose a later version using `--kubernetes-version`

**Comment by bhack** [comment]
I think that we have steel some issue related to `systemd-resolv` in `v1.10.x` expecially for `vm-driver=none`. See https://github.com/kubernetes/kubeadm/issues/845

**Comment by alex-leonhardt** [comment]
ok, so now it's "too old" and has a conflict in `deploy/minikube/k8s_releases.json` - I'll just close this PR as it's not progressing (not all tests run, no idea why) - seems quite a simple change that didn't go through after even 3 weeks.

**Comment by afbjorklund** [comment]
The basic idea (upgrade the kubernetes version) is still valid, even though the Pull Request wasn't.
Note that you can still use the newer versions, and that localkube (k8s_releases.json) is removed.

@tstromberg : something to consider for the next release, perhaps ? (`v1.10.0` is somewhat buggy)

**Comment by afbjorklund** [comment]
@alex-leonhardt : `minikube config set kubernetes-version v1.10.8` (no patches required)

**Pull request closed without merging**